cloud/aws/services/ec2/service.go

```diff
@@ -13,10 +13,22 @@
 
 package ec2
 
+import "github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+
 // Service holds a collection of interfaces.
 // The interfaces are broken down like this to group functions together.
 // One alternative is to have a large list of functions from the ec2 client.
 type Service struct {
 	Instances instances
 	VPCs      vpcs
+	Subnets   subnets
+}
+
+// NewService returns a new service given the ec2 api client.
+func NewService(i ec2iface.EC2API) *Service {
```

Comment:
I see your point, wouldn't this add another layer of abstraction though? Also I'm not really sure about the exposing an internal interface in a public function. The thinking behind this was that for `NewService` a user would either use the mock or the ec2 client.

Reply:
hmm, yeah good call on exposing a private interface. This is already a problem (we exposed private interfaces in a public struct, `Service`) -- I think we should make those internal interfaces public and properly expose them, but that's out of scope. I'll open an issue to track it.

I'm very reluctant to use the defined ec2 api interface since it's big and we really don't need all of it.

I'm ok with this going through as is for now though